### PR TITLE
Issue 39881: User expiration date does not reset after re-activation

### PR DESCRIPTION
**Pull request by labkey-adam**

### CODE REVIEW

File: api/src/org/labkey/api/security/UserManager.java [hunk 1]
```diff
@@ -619,6 +619,13 @@ public static void updateLogin(User user)
         new SqlExecutor(CORE.getSchema()).execute(sql);
     }
 
+    /** Clear the ExpirationDate field for the given user */
+    public static void clearExpirationDate(User currentUser, User user)
```
Comment (Sigmonia) [inline api/src/org/labkey/api/security/UserManager.java]: The param names aren't super clear. Just reading this method I can't tell which user is being updated.
```suggestion
    public static void clearExpirationDate(User executor, User expiringUser)
```

File: api/src/org/labkey/api/audit/AbstractAuditTypeProvider.java [hunk 2]
```diff
@@ -136,8 +134,7 @@ protected void updateIndices(Domain domain, AbstractAuditDomainKind domainKind)
             return;
 
         Map<String, Pair<TableInfo.IndexType, List<ColumnInfo>>> existingIndices = getSchema().getTable(domain.getStorageTableName()).getAllIndices();
-        Set<PropertyStorageSpec.Index> newIndices = new HashSet<>();
-        newIndices.addAll(domainKind.getPropertyIndices(domain));
+        Set<PropertyStorageSpec.Index> newIndices = new HashSet<>(domainKind.getPropertyIndices(domain));
         Set<PropertyStorageSpec.Index> toRemove = new HashSet<>();
```
Comment (Sigmonia) [inline api/src/org/labkey/api/audit/AbstractAuditTypeProvider.java:138]: Is this file & changes part of this fix or just targets of opportunity? seems more like the latter

**Review by Sigmonia — APPROVED** [review]

File: api/src/org/labkey/api/audit/AbstractAuditTypeProvider.java (hunk 2 above)
Comment (labkey-adam) [reply api/src/org/labkey/api/audit/AbstractAuditTypeProvider.java:138]: Yes, opportunistic. Just some fixes to warnings in this file that were hanging out waiting for a FB.

File: api/src/org/labkey/api/security/UserManager.java (hunk 1 above)
Comment (labkey-adam) [reply api/src/org/labkey/api/security/UserManager.java]: Improved in 4c37b77

**Pull request merged**